compose/snippets/src/main/java/com/example/compose/snippets/pictureInPicture/PictureInPictureSnippets.kt

```diff
@@ -0,0 +1,334 @@
+    val context = LocalContext.current
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val pipModifier = modifier.onGloballyPositioned { layoutCoordinates ->
+            val builder = PictureInPictureParams.Builder()
+
+            // Add autoEnterEnabled for versions S and up
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                builder.setAutoEnterEnabled(shouldEnterPipMode)
+            }
+            context.findActivity().setPictureInPictureParams(builder.build())
+        }
+    } else {
+        Log.i("PiP info", "API does not support PiP")
+    }
+    // [END android_compose_pip_post_12_should_enter_pip]
+}
+
+@Composable
+fun PiPBuilderSetSourceRect(
+    shouldEnterPipMode: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    // [START android_compose_pip_set_source_rect]
+    val context = LocalContext.current
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val pipModifier = modifier.onGloballyPositioned { layoutCoordinates ->
+            val builder = PictureInPictureParams.Builder()
+            if (shouldEnterPipMode) {
+                val sourceRect = layoutCoordinates.boundsInWindow().toAndroidRectF().toRect()
+                builder.setSourceRectHint(sourceRect)
+            }
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                builder.setAutoEnterEnabled(shouldEnterPipMode)
+            }
+            context.findActivity().setPictureInPictureParams(builder.build())
+        }
+    } else {
+        Log.i("PiP info", "API does not support PiP")
+    }
+    // [END android_compose_pip_set_source_rect]
+}
+
+@Composable
+fun PiPBuilderSetAspectRatio(
+    shouldEnterPipMode: Boolean,
+    modifier: Modifier = Modifier,
+) {
+    // [START android_compose_pip_set_aspect_ratio]
+    val context = LocalContext.current
+
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+        val pipModifier = modifier.onGloballyPositioned { layoutCoordinates ->
+            val builder = PictureInPictureParams.Builder()
+
+            if (shouldEnterPipMode) {
+                val sourceRect = layoutCoordinates.boundsInWindow().toAndroidRectF().toRect()
+                builder.setSourceRectHint(sourceRect)
+                builder.setAspectRatio(
+                    Rational(sourceRect.width(), sourceRect.height())
+                )
+            }
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+                builder.setAutoEnterEnabled(shouldEnterPipMode)
+            }
+            context.findActivity().setPictureInPictureParams(builder.build())
+        }
+    } else {
+        Log.i("PiP info", "API does not support PiP")
+    }
+    // [END android_compose_pip_set_aspect_ratio]
+}
+
+// [START android_compose_pip_broadcast_receiver]
+@RequiresApi(Build.VERSION_CODES.O)
+@Composable
+fun PlayerBroadcastReceiver(player: Player?) {
+    if (!isInPipMode() || player == null) {
```

Comment:
Since `isInPipMode` internally `remember`s a mutable state, I think the result should be set to a variable to it's clear that a state read is happening. So:

```
val isInPipMode = rememberIsInPipMode()
if (!isInPipMode || player == null) {
    // ...
} 
```

Reply:
Not resolved, I think I agree